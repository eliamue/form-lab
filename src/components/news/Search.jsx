import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ articleTitle, onArticleTitleChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <label htmlFor="articleTitle">Search by Keyword in Title: </label>
    <input
      id="articleTitle"
      type="type"
      value={articleTitle}
      onChange={onArticleTitleChange}
    />
    <button aria-label="find-articles">Submit</button>
  </form>
);

Search.propTypes = {
  articleTitle: PropTypes.string.isRequired,
  onArticleTitleChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default Search;
