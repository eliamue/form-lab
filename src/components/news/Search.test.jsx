import dotenv from 'dotenv';
dotenv.config();
import React from 'react';
import { render, screen } from '@testing-library/react';
import Search from './Search';

describe('Controls', () => {
  it('displays a form', () => {
    render(
      <Search
        articleTitle="Bender"
        onCArticleTitleChange={() => {}}
        onSubmit={() => {}}
      />
    );

    const form = screen.getByTestId('search-form');
    expect(form).toMatchSnapshot();
  });
});
