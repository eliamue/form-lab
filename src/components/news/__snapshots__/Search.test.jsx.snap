// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls displays a form 1`] = `
<form
  data-testid="search-form"
>
  <label
    for="articleTitle"
  >
    Search by Keyword in Title: 
  </label>
  <input
    id="articleTitle"
    type="type"
    value="Bender"
  />
  <button
    aria-label="find-articles"
  >
    Submit
  </button>
</form>
`;
